File: functions/src/constants/TimeZone.ts
export enum TimeZone {
  JST = "Asia/Tokyo", // 日本標準時
  UTC = "UTC", // 協定世界時
  TAIPEI = "Asia/Taipei",
  AMERICAN_EAST = "America/New_York",
  AMERICAN_WEST = "America/Los_Angeles",
  PARIS = "Europe/Paris",
}

export const TriggerTimeZone = TimeZone.JST; // トリガーのタイムゾーン。デフォルトは日本時間

File: app/src/main/java/gaku/original/myapplication/data/Constants/TimeZone.kt
package gaku.original.myapplication.data.Constants

import java.time.ZoneId

/**
 * idを取り出したいときは、TimeZoneOption.JAPAN.id のように使う。
 * labelを取り出したいときは、TimeZoneOption.JAPAN.label のように使う。
 */
enum class TimeZone(val id: String, val label: String) {
    JAPAN("Asia/Tokyo", "Japan"),
    TAIWAN("Asia/Taipei", "Taiwan"),
    EUROPE("Europe/Paris", "Europe"),
    USA("America/New_York", "America East"),
    UTC("UTC", "UTC");

    val zoneId: ZoneId
        get() = ZoneId.of(id)

    companion object {
        fun fromId(id: String): TimeZone? =
            entries.find { it.id == id }
    }

    override fun toString() = label
}